src/main/java/uk/gov/companieshouse/accounts/association/service/AssociationsService.java

```diff
@@ -191,4 +196,42 @@ public void updateAssociation(final String associationId, final Update update) {
 
     }
 
+    private AssociationDao filterForMostRecentInvitation( AssociationDao associationDao ){
+        final var invitations = associationDao.getInvitations();
+
+        if ( invitations.size() > 1 ){
+            final var mostRecentInvitation = invitations.stream()
+                    .max( Comparator.comparing( InvitationDao::getInvitedAt ) )
+                    .orElseThrow();
+
+            invitations.clear();
+            associationDao.setInvitations( List.of( mostRecentInvitation ) );
+        }
+
+        return associationDao;
+    }
+
+    @Transactional( readOnly = true )
+    public List<Invitation> fetchActiveInvitations( final String userId, final int pageIndex, final int itemsPerPage ){
+        return associationsRepository.fetchAssociationsWithActiveInvitations( userId, LocalDateTime.now() )
+                .map( this::filterForMostRecentInvitation )
+                .sorted( ( firstAssociation, secondAssociation ) -> {
+                            final var firstAssociationInvitation = firstAssociation.getInvitations().getFirst().getInvitedAt();
+                            final var secondAssociationInvitation = secondAssociation.getInvitations().getFirst().getInvitedAt();
+                            return secondAssociationInvitation.compareTo( firstAssociationInvitation );
+                        } )
+                .skip((long) pageIndex * itemsPerPage )
+                .limit( itemsPerPage )
+                .flatMap( invitationMapper::daoToDto )
+                .toList();
+    }
+
+    public List<Invitation> fetchInvitations( final AssociationDao associationDao ) {
+        List<Invitation> invitations = associationDao.getInvitations().stream()
```

Comment:
Can make this change, have looked at Google's Java style guide (which companies house uses) and it says [only declare when needed](https://google.github.io/styleguide/javaguide.html#s4.8.2-variable-declarations) - the reason I typically do this is because it's convenient for setting a debug point and looking at the state of the local variable.

Reply:
Have addressed this in https://github.com/companieshouse/accounts-association-api/pull/101/commits/33a3dffacf146595d164e418665b91b36eec3943 - thanks!